shared/teams/team/subteams-tab/helper.js

```diff
@@ -0,0 +1,52 @@
+// @flow
+import * as React from 'react'
+import * as I from 'immutable'
+import * as Constants from '../../../constants/teams'
+import * as Types from '../../../constants/types/teams'
+import {type TypedState} from '../../../util/container'
+import Add from './add-row/container'
+import Intro from './intro-row/container'
+import None from './none-row'
+import Team from './team-row/container'
+
+export type OwnProps = {
+  teamname: string,
+}
+
+type StateProps = {
+  _sawSubteamsBanner: boolean,
+  _subteams: I.Set<Types.Teamname>,
+  _yourOperations: Types.TeamOperations,
+}
+
+export const mapStateHelper = (state: TypedState, {teamname}: OwnProps): StateProps => ({
+  _sawSubteamsBanner: state.teams.getIn(['sawSubteamsBanner'], false),
+  _subteams: Constants.getTeamSubteams(state, teamname),
+  _yourOperations: Constants.getCanPerform(state, teamname),
+})
+
+export const getRows = ({_subteams, _sawSubteamsBanner, _yourOperations}: StateProps) => {
+  const subteams = _subteams.sort()
+  const noSubteams = subteams.isEmpty()
+  return [
```

Comment:
This is kind of hard to read. Can you split it into `const rows = []; if (...) etc.`? Then you can define the `intro`, `add`, and `none` rows as constants somewhere